Channel/DoctrineChannel.php

```diff
@@ -43,20 +46,38 @@ public function supports($recipient)
      */
     public function configure(OptionsResolver $resolver)
     {
+        $resolver
+            ->setDefined(['attachments_path'])
+        ;
     }
 
     /**
      * @inheritdoc
      */
     public function handle(Delivery $delivery)
     {
+        $options = $delivery->getOptions();
+
         $notification = new Notification(
             $delivery->getSubject(),
             $delivery->getBody(),
             $delivery->getRecipient()
         );
 
+        foreach ($delivery->getAttachments() as $attachment) {
+            /** @var $attachment File */
+            $fs = new Filesystem();
+            $fs->copy(
+                $attachment->getPathname(),
+                sprintf('%s/%s', $options['attachments_path'], $attachment->getBasename())
+            );
+            $notificationAttachment = new NotificationAttachment($notification, $attachment->getBasename());
+            $notification->addNotificationAttachment($notificationAttachment);
+
+            $this->manager->persist($notificationAttachment);
```

Comment:
Why persisting the object as standalone? Cascade persist (cascade all in this case) on the relation should be enough